pilot/cmd/pilot-agent/main.go

```diff
@@ -371,7 +396,9 @@ var (
 
 			log.Infof("PilotSAN %#v", pilotSAN)
 
-			envoyProxy := envoy.NewProxy(proxyConfig, role.ServiceNode(), proxyLogLevel, proxyComponentLogLevel, pilotSAN, role.IPAddresses, dnsRefreshRate)
+			envoyProxy := envoy.NewProxy(
```

Comment:
it might be time to make this a struct

Reply:
In a separate PR maybe. This is pretty urgent and should stay focused on solving the startup, not refactor.